foundation/foundation-configuration-store/src/main/java/com/boozallen/aissemble/configuration/mutating/webhook/ConfigMutatingWebhook.java

```diff
@@ -83,9 +87,19 @@ public AdmissionReview validate(AdmissionReview admissionReviewRequest) {
         AdmissionResponse response;
         try {
             String json = mapper.writeValueAsString(request.getObject());
+
+            //This scans to see if there's any config need to be injected and replace corresponding values and saves into updatedJson
             String updatedJson = replaceConfigStoreKeysWithConfigValue(json);
 
-            if (!updatedJson.isEmpty()) {
+            JSONObject jsonObject = !updatedJson.isEmpty() ? new JSONObject(updatedJson) : new JSONObject(json);
```

Comment:
```suggestion
            JSONObject jsonObject = new JSONObject(json);
```
We're just interested in whether the `data` field exists, this shouldn't be affected by injection.

Reply:
if updatedJson have nothing to inject, it would return empty string in line 92, in which could result in nullpointer trying to instantiate JSON object of empty string.

Reply:
This would be using the original Json object so it should never be empty. We don't need to check the updated json for the data field.

Reply:
you're 100% correct, didn't look this throughly updated!